### PR TITLE
Updates for new functionality of the EZO component.

### PR DESCRIPTION
**Pull request by gvdhoven**
## Description:

Updated docs for ezo PR

**Related issue (if applicable):** fixes N/A

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#3561

## Checklist:

  - [X] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`.
  - [X] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

**Comment by Assunto** [comment]
@gvdhoven  I'm trying to set this up and am hitting a confusing snag because the example code from #3561 suggests the way to talk to the sensors is through mqtt where the #2133 deploy preview page makes me think you do it through lambdas? I'm not very good at either.

Would you mind explaining which method to use and a super simple tutorial on getting it set up?

**Comment by gvdhoven** [comment]
> @gvdhoven I'm trying to set this up and am hitting a confusing snag because the example code from #3561 suggests the way to talk to the sensors is through mqtt where the #2133 deploy preview page makes me think you do it through lambdas? I'm not very good at either.
> 
> Would you mind explaining which method to use and a super simple tutorial on getting it set up?

That is out of scope of this PR to be honest..

**Comment by Assunto** [comment]
Is there a reason why this wasn't included in today's esphome update?

**Comment by jkeys1989** [comment]
> [@Assunto](/Assunto) [Assunto](/Assunto) mentioned this pull request [ on Jun 19 ](#ref-pullrequest-1261118784)

I was wondering the same.

**Comment by jncanches** [comment]
Will this PR be included in next release ?
Calibration is a very great update ! Thanks !

**Comment by jkeys1989** [comment]
> Will this PR be included in next release ? Calibration is a very great update ! Thanks !

I'm currently using this like. .. 

works perfect. 

external_components:
  - source: github://pr#3561
    components: [ ezo ]

sensor:

  - platform: ezo
    id: ph_ezo
    address: 99
    accuracy_decimals: 2
    name: "PH Sensor"
    unit_of_measurement: "pH"
    update_interval: 10s

button:

  - platform: template
    name: "Cal 7 PH"
    on_press:
        lambda: !lambda |-
           id(ph_ezo).send_custom("cal,mid,7");

  - platform: template
    name: "Cal 4 PH"
    on_press:
        lambda: !lambda |-
           id(ph_ezo).send_custom("cal,low,4");

  - platform: template
    name: "Cal 10 PH"
    on_press:
        lambda: !lambda |-
           id(ph_ezo).send_custom("cal,high,10");

  - platform: template
    name: "ph led on"
    on_press:
        lambda: !lambda |-
           id(ph_ezo).send_custom("L,1");

File: components/sensor/ezo.rst [hunk 1]
```diff
@@ -9,6 +9,8 @@ EZO sensor circuits
 The ``ezo`` sensor platform allows you to use your EZO sensor circuits with
 ESPHome. The :ref:`I²C Bus <i2c>` is
 required to be set up in your configuration for this sensor to work.
+All embedded solutions from EZO can be found `here <https://atlas-scientific.com/embedded-solutions/>`__.
+If a certain command is not supported by default, it can be executed with
```
Comment (jesserockz) [inline components/sensor/ezo.rst]: Stopped mid sentence?

File: components/sensor/ezo.rst [hunk 2]
```diff
@@ -43,20 +45,146 @@ Configuration variables:
...
+
+Callbacks
+---------
+
+- **on_led:** : Triggered when the result of ``get_led_state()`` is ready
+- **on_device_information:** : Triggered when the result of ``get_device_information()`` is ready
+- **on_slope:** : Triggered when the result of ``get_slope()`` is ready
+- **on_calibration:** : Triggered when the result of ``get_calibration()`` is ready
+- **on_t:** : Triggered when the result of ``get_t()`` is ready
+- **on_custom:** : Triggered when the result of ``get_custom()`` is ready
```
Comment (jesserockz) [inline components/sensor/ezo.rst]: These should be part of the above configuration variables list in the correct format.

**Review by jesserockz — CHANGES_REQUESTED** [review]

**Review by jesserockz — APPROVED** [review]

**Pull request merged**